models/articles-model.js

```diff
@@ -34,3 +35,20 @@ exports.selectAllArticles = () => {
       return articlesArray;
     });
 };
+
+exports.selectComments = (articleId) => {
+  if (!Number(articleId)) {
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
+  return db
+    .query(
+      `SELECT * FROM comments WHERE article_id=${articleId} ORDER BY created_at DESC`
+    )
+    .then((comments) => {
+      const commentsArray = comments.rows;
+      if (commentsArray.length === 0) {
```

Comment:
Is it an error if an article has no comments? What about a new article? No one would have commented on it yet so that wouldn't be an error at all.

This end point does need a 404 so you'll need to consider, does empty rows mean no article or does it mean no comments? And which one is a problem, and which one isn't.

Reply:
<img width="942" alt="Screenshot 2024-08-27 at 13 54 51" src="https://github.com/user-attachments/assets/4e9bfc9a-dccb-4c67-8726-14e462e288f6">
<img width="618" alt="Screenshot 2024-08-27 at 13 55 06" src="https://github.com/user-attachments/assets/1ac30f43-b
<img width="726" alt="Screenshot 2024-08-27 at 13 54 13" src="https://github.com/user-attachments/assets/302be194-f26b-48c8-9793-55f5a5d8e0d7">
849-43a7-a460-cba4894ec6cf">

<img width="1470" alt="Screenshot 2024-08-27 at 13 54 59" src="https://github.com/user-attachments/assets/d0cc967f-1c50-44bd-91f2-710cba94ad5a">

Reply:
<img width="942" alt="Screenshot 2024-08-27 at 13 54 51" src="https://github.com/user-attachments/assets/42386405-7933-4bdf-adf5-6fe76f0b6278">

<img width="1470" alt="Screenshot 2024-08-27 at 13 54 59" src="https://github.com/user-attachments/assets/ac72bb7f-e5a8-43e6-aad4-88f23da3b9b3">

<img width="618" alt="Screenshot 2024-08-27 at 13 55 06" src="https://github.com/user-attachments/assets/06fc49e9-7f66-41a8-809e-9ef483a2d4e5">
<img width="726" alt="Screenshot 2024-08-27 at 13 54 13" src="https://github.com/user-attachments/assets/b3cc8dd0-7c5f-486c-91f1-f10cff8dc869">